casper-private-auction-core/src/auction.rs

```diff
@@ -87,7 +93,7 @@ impl Auction {
     }
 
     fn auction_transfer_token(recipient: Key) {
-        let auction_key: Key = {
```

Comment:
So, this is just a call from the _auction_ to the NFT contract, and it obtains the auction's contract package hash in a very weird and roundabout manner, so this just needs simplification (maybe just store the auction's contract hash in a named key)

Reply:
Will try that

Reply:
Hi, I am going to submit [that change](https://github.com/gRoussac/casper-private-auction/pull/3/commits/4158a2b5be57b7f1f2d19a11c0db07cdc18da224) in a second PR.